src/Tutor.Core/DomainModel/KnowledgeComponents/AssessmentEventHelp/AssessmentEventHelpService.cs

```diff
@@ -0,0 +1,34 @@
+﻿using FluentResults;
+using Tutor.Core.DomainModel.AssessmentEvents;
+
+namespace Tutor.Core.DomainModel.KnowledgeComponents.AssessmentEventHelp
+{
+    public class AssessmentEventHelpService : IAssessmentEventHelpService
+    {
+        private readonly IAssessmentEventRepository _assessmentEventRepository;
+        private readonly IKCRepository _kcRepository;
+
+        public AssessmentEventHelpService(IAssessmentEventRepository assessmentEventRepository, IKCRepository kcRepository)
+        {
+            _assessmentEventRepository = assessmentEventRepository;
+            _kcRepository = kcRepository;
+        }
+
+        public Result SeekHelp(int learnerId, int assessmentEventId, string helpType)
+        {
+            var assessmentEvent = _assessmentEventRepository.GetDerivedAssessmentEvent(assessmentEventId);
```

Comment:
Do we need to load the whole assessment event (as it has many joins) or can we add a repository function that retrieves the KcID for the AEid?

Reply:
I added a method to retrieve KcMastery by the AeId and LearnerId in [20803ba](https://github.com/Clean-CaDET/tutor/pull/26/commits/20803ba0cd8ef34974d76b85ab10d3b6c3060eb6). Using that method in AeHelpService is fine since it doesn't actually use the AE, but in SubmissionService the retrieval had to stay as it was (reverted in [049a7be](https://github.com/Clean-CaDET/tutor/pull/26/commits/049a7bebb1a722fe9695aaebec1dffddcb1dc36a)) because it actually needs the entire AE.

Reply:
It makes sense to add this Repo class and feel free to introduce some repository refactoring along the way.